Code builds with warnings v1.0.3

I'm trying to update the cyw43-driver in pico-sdk. We have a test that everything builds without warnings and errors. There are some new issues in the latest version of the cyw43_driver v1.03.

```
/home/peterh/source/pico/pico-sdk2/lib/cyw43-driver/src/cyw43_ll.c: In function 'cyw43_check_valid_chipset_firmware':
/home/peterh/source/pico/pico-sdk2/lib/cyw43-driver/src/cyw43_ll.c:367:60: error: unused parameter 'self' [-Werror=unused-parameter]
  367 | static int cyw43_check_valid_chipset_firmware(cyw43_int_t *self, size_t len, uintptr_t source) {
      |                                               ~~~~~~~~~~~~~^~~~
In file included from /home/peterh/source/pico/pico-sdk2/lib/cyw43-driver/src/cyw43_ll.c:40:
/home/peterh/source/pico/pico-sdk2/lib/cyw43-driver/src/cyw43_ll.c: In function 'cyw43_ll_sdpcm_poll_device':
/home/peterh/source/pico/pico-sdk2/lib/cyw43-driver/src/cyw43_ll.c:1097:25: error: format '%u' expects argument of type 'unsigned int', but argument 2 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]
 1097 |             CYW43_DEBUG("SPI invalid bytes pending %u\n", bytes_pending);
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~
      |                                                           |
      |                                                           uint32_t {aka long unsigned int}
/home/peterh/source/pico/pico-sdk2/lib/cyw43-driver/src/cyw43_config.h:122:34: note: in definition of macro 'CYW43_PRINTF'
  122 | #define CYW43_PRINTF(...) printf(__VA_ARGS__)
      |                                  ^~~~~~~~~~~
/home/peterh/source/pico/pico-sdk2/lib/cyw43-driver/src/cyw43_ll.c:1097:13: note: in expansion of macro 'CYW43_DEBUG'
 1097 |             CYW43_DEBUG("SPI invalid bytes pending %u\n", bytes_pending);
      |             ^~~~~~~~~~~
/home/peterh/source/pico/pico-sdk2/lib/cyw43-driver/src/cyw43_ll.c:1097:53: note: format string is defined here
 1097 |             CYW43_DEBUG("SPI invalid bytes pending %u\n", bytes_pending);
      |                                                    ~^
      |                                                     |
      |                                                     unsigned int
      |                                                    %lu
cc1: all warnings being treated as errors
```

--- a/src/cyw43_ll.c
+++ b/src/cyw43_ll.c
@@ -365,6 +365,7 @@ static void cyw43_write_backplane(cyw43_int_t *self, uint32_t addr, size_t size,
 }
 
 static int cyw43_check_valid_chipset_firmware(cyw43_int_t *self, size_t len, uintptr_t source) {
+    (void)self;
     // get the last bit of the firmware, the last 800 bytes
     uint32_t fw_end = 800;
     const uint8_t *b = (const uint8_t *)source + len - fw_end;
@@ -1094,7 +1095,7 @@ static int cyw43_ll_sdpcm_poll_device(cyw43_int_t *self, size_t *len, uint8_t **
         bytes_pending = (bus_gspi_status >> 9) & 0x7FF;
         if (bytes_pending == 0 || bytes_pending > (LINK_MTU - GSPI_PACKET_OVERHEAD) ||
             bus_gspi_status & F2_F3_FIFO_RD_UNDERFLOW) {
-            CYW43_DEBUG("SPI invalid bytes pending %u\n", bytes_pending);
+            CYW43_DEBUG("SPI invalid bytes pending %" PRIu32 "\n", bytes_pending);
             cyw43_write_reg_u8(self, BACKPLANE_FUNCTION, SPI_FRAME_CONTROL, (1 << 0));
             self->had_successful_packet = false;
             return -1;

